Fresh Kali Linux 2020.1 install fails.

Attempted to install on fresh kali linux 2020.1 VMWare build running as non-root user. 
First ran apt-get update/upgrade/dist-upgrade 
Then cloned git repo and ran setup. 

Both HostAPD and HCXTools failed to compile as per the following logs:

```
[*] Compiling hostapd...

make: pkg-config: Command not found
make: pkg-config: Command not found
In file included from main.c:35:
/home/kali/eaphammer/local/hostapd-eaphammer/src/eaphammer_wpe/eaphammer_wpe.h:14:10: fatal error: openssl/ssl.h: No such file or directory
   14 | #include <openssl/ssl.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:1301: main.o] Error 1

[*] complete!


[*] Compiling hcxtools...

mkdir -p .deps
cc -O3 -Wall -Wextra -std=gnu99  -MMD -MF .deps/hcxpcaptool.d -o hcxpcaptool hcxpcaptool.c -lz -lcrypto 
hcxpcaptool.c:16:10: fatal error: openssl/sha.h: No such file or directory
   16 | #include <openssl/sha.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:81: hcxpcaptool] Error 1

[*] complete!
```

The problem appears to be resolved by

`sudo apt-get install hostapd hcxtools
`

Also, when launching eaphammer for the first time, the pem module was missing from python as shown here:

`**Describe the bug**
Attempted to install on fresh kali linux 2020.1 VMWare build running as non-root user. 
First ran apt-get update/upgrade/dist-upgrade 
Then cloned git repo and ran setup. 

Both HostAPD and HCXTools failed to compile as per the following logs:

```
[*] Compiling hostapd...

make: pkg-config: Command not found
make: pkg-config: Command not found
In file included from main.c:35:
/home/kali/eaphammer/local/hostapd-eaphammer/src/eaphammer_wpe/eaphammer_wpe.h:14:10: fatal error: openssl/ssl.h: No such file or directory
   14 | #include <openssl/ssl.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:1301: main.o] Error 1

[*] complete!


[*] Compiling hcxtools...

mkdir -p .deps
cc -O3 -Wall -Wextra -std=gnu99  -MMD -MF .deps/hcxpcaptool.d -o hcxpcaptool hcxpcaptool.c -lz -lcrypto 
hcxpcaptool.c:16:10: fatal error: openssl/sha.h: No such file or directory
   16 | #include <openssl/sha.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:81: hcxpcaptool] Error 1

[*] complete!
```

The problem appears to be resolved by

`sudo apt-get install hostapd hcxtools
`

Also, when launching eaphammer for the first time, the pem module was missing from python

```
kali@kali:~/eaphammer$ ./eaphammer 
Traceback (most recent call last):
  File "./eaphammer", line 4, in <module>
    import cert_wizard
  File "/home/kali/eaphammer/cert_wizard/__init__.py", line 1, in <module>
    from cert_wizard.core import import_cert
  File "/home/kali/eaphammer/cert_wizard/core.py", line 1, in <module>
    import cert_wizard.cert_utils as cert_utils
  File "/home/kali/eaphammer/cert_wizard/cert_utils.py", line 3, in <module>
    import pem
ModuleNotFoundError: No module named 'pem'
```

As pip/pip3 is not installed in Kali Linux by default, first installing them is required

`sudo apt install python-pip python3-pip`

then installing the missing module.

`pip3 install pem`

```

kali@kali:~/eaphammer$ uname -a
Linux kali 5.5.0-kali2-amd64 #1 SMP Debian 5.5.17-1kali1 (2020-04-21) x86_64 GNU/Linux

```

Hope this is of help to someone else.

--- a/kali-dependencies.txt
+++ b/kali-dependencies.txt
@@ -7,3 +7,4 @@ libnl-genl-3-dev
 libcurl4-openssl-dev
 zlib1g-dev
 libpcap-dev
+python3-pip

--- a/kali-setup
+++ b/kali-setup
@@ -1,13 +1,21 @@
 #!/usr/bin/env python3
 import os
+import sys
 
 from settings import settings
+
+def exit_if_not_root():
+
+    if os.getuid() != 0:
+        sys.exit("[!} Error: this script must be run as root.")
 
 def read_deps_file(deps_file):
     with open(deps_file) as fd:
         return ' '.join([ line.strip() for line in fd ])
 
 if __name__ == '__main__':
+
+    exit_if_not_root()
                     
 
     default_wordlist = os.path.join(settings.dict['paths']['directories']['wordlists'], settings.dict['core']['eaphammer']['general']['default_wordlist'])
@@ -25,19 +33,10 @@ if __name__ == '__main__':
     openssl_bin = settings.dict['paths']['openssl']['bin']
     dh_file = settings.dict['paths']['certs']['dh']
 
-    if input('Do you want update your package list ("apt -y update")(recommended)? [y/N]: ').lower() == 'y':
-        print('\n[*] Downloading package lists from kali repositories...\n')
-        os.system('apt -y update')
-        print('\n[*] complete!\n')
-
+    if input('Important: it is highly recommended that you run "apt -y update" and "apt -y upgrade" prior to running this setup script. Do you wish to proceed? Enter [y/N]: ').lower() != 'y':
+        sys.exit('Aborting.')
     print()
 
-    if input('Do you want upgrade your installed packages ("apt -y upgrade")(recommended)? [y/N]: ').lower() == 'y':
-        print('\n[*] Performing upgrade...\n')
-        os.system('apt -y upgrade')
-        print('\n[*] complete!\n')
-
-    print()
 
     print('\n[*] Removing stub files...\n')
     os.system('find {} -type f -name \'stub\' -exec rm -f {{}} +'.format(root_dir))

